packages/core/src/implementations/business/utilities/QueryParsingEngine.ts

```diff
@@ -222,7 +223,10 @@ export class QueryParsingEngine implements IQueryParsingEngine {
     | MethodSupportError
     | InvalidParametersError
   > {
-    return this.handleQuery(query, DataPermissions.createWithAllPermissions());
+    return this.handleQuery(
+      query,
+      new DataPermissions("" as EVMContractAddress, [], []),
+    ); // @TODO we should probably use real permissions here
```

Comment:
Agreed. Please fix up this TODO. This is pretty straightforward, `this.permissionRepo.getPermissions(query.consentContractAddress).andThen((perms) => {return this.handlQuery(query, perms);});`